### PR TITLE
Bump foundry to 0.28.0

### LINKED ISSUE #1111: Update snforge to 0.28.0
[This sn-foundry release](https://github.com/foundry-rs/starknet-foundry/releases/tag/v0.28.0) introduces a `DeclareResult` return value for the declare cheatcode, which allows us to refactor the declare and deploy functions under the testing package.

### PR DESCRIPTION
**Pull request by ericnordelo**
Fixes #1111 

#### PR Checklist

- [x] Tests
- [ ] Documentation
- [x] Added entry to CHANGELOG.md 
- [ ] Tried the feature on a public network

### CODE REVIEW

File: docs/modules/ROOT/pages/api/testing.adoc [hunk 1]
```diff
@@ -120,6 +120,9 @@ A module containing utilities that simplify declaring and deploying contracts us
 
 Declares a contract with a `snforge` `declare` call and unwraps the result.
 
+NOTE: If the contract is already declared, this function won't panic but will instead
+return the existing class.
+
```
Comment (immrsd) [inline docs/modules/ROOT/pages/api/testing.adoc:125]: Great news!

File: packages/account/src/tests/test_account.cairo [hunk 2]
```diff
@@ -50,7 +52,7 @@ fn setup(key_pair: StarkKeyPair) -> ComponentState {
 fn setup_dispatcher(
     key_pair: StarkKeyPair, data: SignedTransactionData
 ) -> (AccountABIDispatcher, felt252) {
-    let contract_class = declare("DualCaseAccountMock").unwrap_syscall();
+    let contract_class = *declare("DualCaseAccountMock").unwrap_syscall().contract_class();
```
Comment (immrsd) [inline packages/account/src/tests/test_account.cairo:55]: We could use `utils::declare_class` here, that would make code a bit nicer and we wouldn't have to import `DeclareResultTrait`

File: packages/token/src/tests/erc1155/test_erc1155.cairo [hunk 3]
```diff
@@ -1028,9 +1028,7 @@ fn test_update_wac_batch_from_non_zero_to_non_zero_account() {
 
 #[test]
 #[should_panic(
-    expected: (
-        "Contract not deployed at address: 0x0000000000000000000000000000000000000000000000000000000000000000",
-    )
+    expected: "Contract not deployed at address: 0x0000000000000000000000000000000000000000000000000000000000000000"
```
Comment (immrsd) [inline packages/token/src/tests/erc1155/test_erc1155.cairo:1031]:
```suggestion
    expected: "Contract not deployed at address: 0x0"
```

**Comment by ericnordelo** [comment]
Since the release has [some issues](https://github.com/foundry-rs/starknet-foundry/releases/tag/v0.28.0) with test collection, let's wait for the fix before updating and merging this PR.

**Comment by ericnordelo** [comment]
Closing in favor of #1137

**Pull request closed without merging**